Java/libraries/recognizers-text-datatypes-timex-expression/src/main/java/com/microsoft/recognizers/datatypes/timex/expression/TimexHelpers.java

```diff
@@ -178,6 +179,31 @@ public static TimexProperty timexDateAdd(TimexProperty start, TimexProperty dura
         return start;
     }
 
+    public static String generateDateTimex(Integer year, Integer monthOrWeekOfYear, Integer day, Integer weekOfMonth, boolean byWeek)
+    {
+        String yearString = year == Constants.InvalidValue ? Constants.TimexFuzzyYear : TimexDateHelpers.fixedFormatNumber(year, 4);
+        String monthWeekString = monthOrWeekOfYear == Constants.InvalidValue ? Constants.TimexFuzzyMonth : TimexDateHelpers.fixedFormatNumber(monthOrWeekOfYear, 2);
+        String dayString;
+        if (byWeek)
+        {
+            dayString = day.toString(Integer.parseInt(Locale.ROOT.getLanguage()));
+            if (weekOfMonth != Constants.InvalidValue)
+            {
+                monthWeekString = monthWeekString + String.format(Constants.TimexFuzzyWeek, weekOfMonth.toString(Integer.parseInt(Locale.ROOT.getLanguage())));
```

Comment:
```suggestion
                monthWeekString = monthWeekString + String.format(-%s-, Constants.TimexFuzzyWeek) weekOfMonth.toString(Integer.parseInt(Locale.ROOT.getLanguage()));
```

Reply:
Also not sure about `toString(Integer.parseInt(Locale.ROOT.getLanguage()))`

Reply:
If I'm not mistaken, the toString isn't even necessary on the weekOfMonth. The String.format method takes an array of Object

Reply:
If that's the case, lgtm